src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutConnectorCheck.java

```diff
@@ -0,0 +1,156 @@
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        // Get the object as an Edge
+        final Edge edge = (Edge) object;
+        // Look for a connected roundabout Edge
+        final Optional<Edge> optionalRoundaboutEdge = this.getRoundaboutEdge(edge);
+        // Short circuit if the edge is not connected to a roundabout
+        if (!optionalRoundaboutEdge.isPresent())
+        {
+            return Optional.empty();
+        }
+        final Edge roundaboutEdge = optionalRoundaboutEdge.get();
+
+        String instruction = null;
+
+        // Get the correct heading depending on if we are entering or exiting the roundabout and
+        // record the connecting Node
+        final Optional<Heading> edgeHeading;
+        final Optional<Heading> roundaboutHeading;
+        if (edge.inEdges().contains(roundaboutEdge))
+        {
+            edgeHeading = edge.asPolyLine().initialHeading();
+            roundaboutHeading = roundaboutEdge.asPolyLine().finalHeading();
+        }
+        else
+        {
+            edgeHeading = edge.asPolyLine().finalHeading();
+            roundaboutHeading = roundaboutEdge.asPolyLine().initialHeading();
+        }
+
+        // Check that we have both headings
+        if (edgeHeading.isPresent() && roundaboutHeading.isPresent())
+        {
+            // If this is a two way road and the angle is greater than the two way threshold...
+            if (edge.hasReverseEdge() && edgeHeading.get().difference(roundaboutHeading.get())
+                    .isGreaterThanOrEqualTo(this.twoWayThreshold))
+            {
+                instruction = this.getLocalizedInstruction(1, object.getOsmIdentifier());
+
+            }
+            // If this is one way road and the angle is greater than the one way threshold...
+            else if (!edge.hasReverseEdge() && edgeHeading.get().difference(roundaboutHeading.get())
+                    .isGreaterThanOrEqualTo(this.oneWayThreshold))
+            {
+                instruction = this.getLocalizedInstruction(0, object.getOsmIdentifier());
+            }
+        }
+
+        if (instruction != null)
+        {
+            this.markAsFlagged(edge.getOsmIdentifier());
+            return Optional.of(this.createFlag(new OsmWayWalker(edge).collectEdges(), instruction));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Get the first connected roundabout {@link Edge} from the in and out {@link Edge}s
+     *
+     * @param edge
+     *            the {@link Edge} to look for connected roundabouts from
+     * @return an {@link Optional} of a roundabout {@link Edge}
+     */
+    private Optional<Edge> getRoundaboutEdge(final Edge edge)
+    {
+        return Stream.concat(edge.inEdges().stream(), edge.outEdges().stream())
```

Comment:
Why don't we just use edge.connectedEdges() here instead of concatenating the in and out edges?

Reply:
Connected Edges can contain edges that route into the end node or out of the start node. These are not navigable connected, and thus we don't want to consider them.

Reply:
Makes sense. Thanks!